src/sudoku/sudoku.py

```
@@ -12,32 +13,35 @@ def __init__(self, grid: List[List[int | None]]) -> None:
         self.grid = grid
         self.n = len(grid)
 
-    # TODO: Change all code to use n>9
     @staticmethod
     def read_sudoku(path: Path) -> "Sudoku":
         grid: List[List[int | None]] = []
         with open(path) as file:
-            for index, line in enumerate(
-                file
-            ):  # Use enumerate to get the index of each line
+            size = 0
+            for index, line in enumerate(file):
                 split: List[str] = line.split()
                 row: List[int | None] = [
                     int(x) if x.isdigit() else None for x in split
                 ]
-                if len(row) < 9:
+                size = max(size, len(row))
+                if math.isqrt(len(row)) ** 2 != len(row):
```

Ah, yes, we didn't talk about this and I could have been clearer what I want. When you are writing error handling code there are two approaches:
- just fix the error and move on or (essentially what you are doing)
- raise an error so someone else can deal with it

I would argue that just extending the grid is probably never what the user wants. Neither is replacing non number values with None. Instead I would raise an error. For this to be usable in a GUI, there is also no point in printing to the command line, because this will not be shown to the user.

So "the proper way" to deal with errors here would be:
- Create an exception base class for all exceptions your sudoku library throws.
- Maybe: add more fine grained subclasses
- Catch the exception in the CLI and print an error message to the user.